### ISSUE TITLE
Add hooks for pre- and post-processing

### ISSUE THREAD

**Issue by madig** [googlei18n/ufo2ft#181]
Working on the Ubuntu font with its' special requirements made me think about pre-/post-processing hooks for fontmake or ufo2ft. The goal is to write custom code that takes care of things like reading in the `maxp` table data present in the UFO or adding a legacy kerning table to the compiled TTF, all without compromising fontmake's simplicity.

**Comment by behdad** [comment]
For things that are commonly used by multiple designers, we should make them available in fontmake itself.

**Comment by madig** [comment]
That would need more designer input first.

I'm thinking about all the fonts out there that were not build with a modern toolset and have special requirements for building (like Adam Twardoch says about Lato: https://github.com/google/fonts/issues/6).

A pre-processor hook would be called after the source (UFO) has been loaded, a post-processor would be able to work on the compiled TTF/OTF in-memory before it is written to disk.

**Comment by behdad** [comment]
Right.  I deliberately didn't say these hooks are not a good idea.

**Comment by madig** [comment]
The hooks have to work such that external tools can work on the fonts, so they'd have to be written out to disk before that and then read back in. In e.g. Ubuntu, the final binaries might be worked on by CacheTT.exe through wine with another Python script doing post-processing after that.
